dashboards-observability/public/components/visualizations/charts/bar/bar.tsx

```diff
@@ -82,43 +90,44 @@ export const Bar = ({ visualizations, layout, config }: any) => {
   if (valueForXSeries.some((e) => e.type === 'timestamp')) {
     const nameData =
       valueForXSeries.length > 1
-        ? (valueForXSeries
-              .filter((item) => item.type !== 'timestamp')
-              .map((dimension) => data[dimension.label])
-          ).reduce((prev, cur) => {
-            return prev.map((i, j) => `${i}, ${cur[j]}`);
-          })
+        ? valueForXSeries
+            .filter((item) => item.type !== 'timestamp')
+            .map((dimension) => data[dimension.label])
+            .reduce((prev, cur) => {
+              return prev.map((i, j) => `${i}, ${cur[j]}`);
+            })
         : [];
 
-    let dimensionsData = 
-      valueForXSeries
-        .filter((item) => item.type === 'timestamp')
-        .map((dimension) => data[dimension.label]).flat();
+    let dimensionsData = valueForXSeries
+      .filter((item) => item.type === 'timestamp')
+      .map((dimension) => data[dimension.label])
+      .flat();
 
-    bars = (valueSeries.map((field: any, index: number) => {
-      const selectedColor = getSelectedColorTheme(field, index);
-      return dimensionsData.map((dimension: any, j: number) => {
-        return {
-          x: isVertical
-            ? !isEmpty(xaxis)
-              ? dimension
-              : data[fields[lastIndex].name]
-            : data[field.label],
-          y: isVertical ? data[field.label][j] : dimensionsData, // TODO: orinetation
-          type: vis.type,
-          marker: {
-            color: hexToRgb(selectedColor, fillOpacity),
-            line: {
-              color: selectedColor,
-              width: lineWidth,
+    bars = valueSeries
```

Comment:
Please check with the developers who wrote this code. If they do not want the `bar` value to be computed on every render, then try using `useMemo`.